pkg/operation/botanist/component/nodelocaldns/nodelocaldns.go

```diff
@@ -0,0 +1,561 @@
+			},
+			AutomountServiceAccountToken: pointer.Bool(false),
+		}
+
+		podSecurityPolicy = &policyv1beta1.PodSecurityPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "gardener.kube-system.node-local-dns",
+				Labels: map[string]string{
+					v1beta1constants.LabelApp: labelValue,
+				},
+			},
+			Spec: policyv1beta1.PodSecurityPolicySpec{
+				AllowedHostPaths: []policyv1beta1.AllowedHostPath{
+					{
+						PathPrefix: "/run/xtables.lock",
+					},
+				},
+				FSGroup: policyv1beta1.FSGroupStrategyOptions{
+					Rule: policyv1beta1.FSGroupStrategyRunAsAny,
+				},
+				HostNetwork: true,
+				HostPorts: []policyv1beta1.HostPortRange{
+					{
+						Min: int32(53),
+						Max: int32(53),
+					},
+					{
+						Min: prometheusPort,
+						Max: prometheusPort,
+					},
+				},
+				Privileged: true,
+				RunAsUser: policyv1beta1.RunAsUserStrategyOptions{
+					Rule: policyv1beta1.RunAsUserStrategyRunAsAny,
+				},
+				SELinux: policyv1beta1.SELinuxStrategyOptions{
+					Rule: policyv1beta1.SELinuxStrategyRunAsAny,
+				},
+				SupplementalGroups: policyv1beta1.SupplementalGroupsStrategyOptions{
+					Rule: policyv1beta1.SupplementalGroupsStrategyRunAsAny,
+				},
+				Volumes: []policyv1beta1.FSType{
+					"secret",
+					"hostPath",
+					"configMap",
+				},
+			},
+		}
+
+		clusterRole = &rbacv1.ClusterRole{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "gardener.cloud:psp:kube-system:node-local-dns",
+				Labels: map[string]string{
+					v1beta1constants.LabelApp: labelValue,
+				},
+			},
+			Rules: []rbacv1.PolicyRule{
+				{
+					APIGroups:     []string{"policy", "extensions"},
+					ResourceNames: []string{podSecurityPolicy.Name},
+					Resources:     []string{"podsecuritypolicies"},
+					Verbs:         []string{"use"},
+				},
+			},
+		}
+
+		roleBinding = &rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "gardener.cloud:psp:node-local-dns",
+				Namespace: metav1.NamespaceSystem,
+				Labels: map[string]string{
+					v1beta1constants.LabelApp: labelValue,
+				},
+				Annotations: map[string]string{resourcesv1alpha1.DeleteOnInvalidUpdate: "true"},
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: rbacv1.GroupName,
+				Kind:     "ClusterRole",
+				Name:     clusterRole.Name,
+			},
+			Subjects: []rbacv1.Subject{{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      serviceAccount.Name,
+				Namespace: serviceAccount.Namespace,
+			}},
+		}
+
+		configMap = &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "node-local-dns",
+				Namespace: metav1.NamespaceSystem,
+			},
+			Data: map[string]string{
+				configDataKey: domain + `:53 {
+  errors
+  cache {
+          success 9984 30
+          denial 9984 5
+  }
+  reload
+  loop
+  bind ` + c.bindIP() + `
+  forward . ` + c.values.ClusterDNS + ` {
+          ` + c.forceTcpToClusterDNS() + `
+  }
+  prometheus :` + strconv.Itoa(prometheusPort) + `
+  health ` + IPVSAddress + `:` + strconv.Itoa(livenessProbePort) + `
+}
```

Comment:
Can't we "fix" those nits even now? Sonu has validated that it works, I think. :)

Reply:
WDYM? The indentation is quite inconsistent in both master and this PR (first items indented with 2/4 spaces, then inner ones - with 8 spaces). @ScheererJ @DockToFuture can you please share how the desired indentation for this ConfigMap should look like?

Reply:
The inconsistent indentation comes originally from https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/nodelocaldns/nodelocaldns.yaml, which uses 4 leading spaces and additional 8 spaces for the next level.
As far as I understand the parsing logic, caddyfiles do not really care about leading spaces. (The CoreDNS configuration file Corefile is a caddyfile.) Therefore, we could stay consistent to how it is formatted in the kubernetes addons guide or change it to "prettier" variant.
As it will be more difficult to `diff` the changes going forward, I do not really care about the indentation, but if pressed for an opinion I would keep it the way it was.

Reply:
I will stick with the original implementation.